### PR TITLE
Better error message for simplification with multiple results

### PR DESCRIPTION
**Pull request by yzhang90**
---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Step/Axiom/EvaluationStrategy.hs [hunk 1]
```diff
@@ -239,10 +239,21 @@ applyFirstSimplifierThatWorks
               error
                   (  "Unexpected simplification result with more "
                   ++ "than one configuration: "
-                  ++ show applicationResult
+                  ++ (show . Pretty.vsep $
+                        [ "Simplification result with remainder:"
```
Comment (ttuegel) [inline kore/src/Kore/Step/Axiom/EvaluationStrategy.hs]: Please change this to use the error message above, i.e. 
```suggestion
                        [ "Unexpected simplification result with more than one configuration:"
```

File: kore/src/Kore/Step/Axiom/EvaluationStrategy.hs [hunk 2]
```diff
@@ -238,11 +238,21 @@ applyFirstSimplifierThatWorks
               -- than one result, so we throw an error.
               error
                   (  "Unexpected simplification result with more "
-                  ++ "than one configuration: "
-                  ++ show applicationResult
+                  ++ "than one configuration: \n"
+                  ++ (show . Pretty.vsep $
```
Comment (yzhang90) [inline kore/src/Kore/Step/Axiom/EvaluationStrategy.hs]: should i use `Pretty.renderStrict . Pretty.layoutCompact` before `Pretty.vsep` like the code below?

Comment (ttuegel) [reply kore/src/Kore/Step/Axiom/EvaluationStrategy.hs]: No, `show` should be fine.

File: kore/src/Kore/Step/Axiom/EvaluationStrategy.hs (hunk 1 above)
Comment (yzhang90) [reply kore/src/Kore/Step/Axiom/EvaluationStrategy.hs]: done

**Pull request merged**